boa/src/syntax/ast/token.rs

```diff
@@ -239,6 +239,8 @@ pub enum TokenKind {
     /// A string literal.
     StringLiteral(Box<str>),
 
+    TemplateLiteral(Box<str>),
```

Comment:
This would need some documentation. Ideally also a link to the spec.

Reply:
Not sure what you mean.

Reply:
At least, the variant should be documented like this:
```suggestion
    /// A template literal.
    TemplateLiteral(Box<str>),
```

Reply:
Commenting a variable named TemplateLiteral with a commend saying "template literal" is bad practice. I guess the comment with the possible link to the spec can go in boa/src/syntax/lexer/mod.rs. I put it there if you check that file in this PR.